Glitter/Sources/main.cpp

```diff
@@ -49,37 +52,170 @@ const GLchar* fragmentSource =
+  //Triangle 2
+  -0.5f, -0.5f,  0.5f, RED,     //Bottom Left (front)
+  -0.5f,  0.5f,  0.5f, BLUE,    //Top Left (front)
+  -0.5f, -0.5f, -0.5f, GREEN,   //Bottom Left (back)
+
+  //Top
+  //Triangle 1
+   0.5f,  0.5f, -0.5f, RED,     // Top Right (back)
+   0.5f,  0.5f,  0.5f, BLUE,    // Top Right (front)
+  -0.5f,  0.5f, -0.5f, GREEN,   // Top Left (back)
+  //Triangle 2
+  -0.5f, 0.5f,  0.5f,  RED,      //Top Left (front)
+  -0.5f, 0.5f, -0.5f,  BLUE,     //Top Left (back)
+   0.5f, 0.5f,  0.5f,  GREEN,    //Top Right (front)
+
+  //Bottom
+  //Triangle 1
+   0.5f, -0.5f, -0.5f, RED,     // Bottom Right (back)
+   0.5f, -0.5f,  0.5f, BLUE,    // Bottom Right (front)
+  -0.5f, -0.5f, -0.5f, GREEN,   // Bottom Left (back)
+  //Triangle 2
+  -0.5f, -0.5f,  0.5f, RED,     //Bottom Left (front)
+  -0.5f, -0.5f, -0.5f, BLUE,    //Bottom Left (back)
+  0.5f,  -0.5f,  0.5f, GREEN,   //Bottom Right (front)
 };
 
+//Camera vectors
+glm::vec3 eye  = glm::vec3(0.0f, 0.0f, 3.0f);
+glm::vec3 gaze = glm::vec3(0.0f, 0.0f, 0.0f);
+glm::vec3 up   = glm::vec3(0.0f, 0.1f, 0.0f);
+
 bool controlIsPressed(GLFWwindow* window) {
-	return glfwGetKey(window, GLFW_KEY_LEFT_CONTROL) == GLFW_PRESS || glfwGetKey(window, GLFW_KEY_RIGHT_CONTROL) == GLFW_PRESS; 
+	return glfwGetKey(window, GLFW_KEY_LEFT_CONTROL) == GLFW_PRESS || glfwGetKey(window, GLFW_KEY_RIGHT_CONTROL) == GLFW_PRESS;
+}
+
+//Prints the current coordinates of a given vector
+void printCurrentCoordinates(glm::vec3 vec){
+  std::cout << "(" << vec.x << ", "
+                   << vec.y << ", "
+                   << vec.z << ")" << std::endl;
 }
 
 // callback for keyboard input
 // move camera when arrow keys are pressed, rotate it when arrow keys are pressed with control
 void key_callback(GLFWwindow* mWindow, int key, int scancode, int action, int mods)
 {
-  std::cout << (mods == GLFW_MOD_SHIFT) << std::endl;
+  GLfloat speed = 0.05f;
+  //When you press shift, it will print out to the console informing you are pressing Shift otherwise it will be null
+  std::cout << ((mods == GLFW_MOD_SHIFT) ? "You Pressed Shift!" : "") << std::endl;
+
   if (key == GLFW_KEY_LEFT && action != GLFW_RELEASE) {
-    if (mods == GLFW_MOD_SHIFT) std::cout << "rotating camera left\n"; // rotate camera
-    else std::cout << "moving camera left\n";  // move camera
+    if (mods == GLFW_MOD_SHIFT){
+      std::cout << "rotating camera left; new coordinates are: "; // rotate camera
+      //In order to rotate the camera left you change x and z (positively)
+      eye.x += speed;
+      eye.z += speed;
+      printCurrentCoordinates(eye);
+    }
+    else{
+      std::cout << "moving camera left; new coordinates are: ";  // move camera
+      gaze.x += speed;
+      printCurrentCoordinates(gaze);
+    }
+  }
     // etc.
+  else if(key == GLFW_KEY_RIGHT && action != GLFW_RELEASE){
+    if(mods == GLFW_MOD_SHIFT){
+      std::cout << "rotating camera right; new coordinates are: ";
```

Comment:
this moves the object left and the camera right (same for all your camera movements -- they are in the opposite directions)